Check can-i-merge on each PR

pjlee11 opened the pull request:
This is the outcome of a learning day.

Resolves https://github.com/bbc/simorgh-infrastructure/issues/143

_Adds a new Jenkinsfile and a bash script for the new `can-i-merge` Jenkins job_

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

Commit:
A jenkinsfile with some Bash

- Create a new directory "can-i-merge" which is used by the Jenkins job "can-i-merge"
- Defines a Jenkinsfile which triggers the bash script
- The bash script does the following:
  * curl's the lastBuild of the simorgh-infrastructure Jenkins job. This job uses CD to put our code live following a github merge
  * If the curl output does not explicitly contain the string `result":"SUCCESS` the job fails
  * Once integrated with the simorgh application this should prohibit merging when the deployment pipeline is not green

Commit:
Merge branch 'latest' of github.com:BBC-News/simorgh into testy-testy

Commit:
\n is not needed on jenkins

Commit:
package-lock shouldn't change in this PR

Commit:
commit npm install for green tick

bcmn commented on can-i-merge/bash.sh:
```diff
@@ -0,0 +1,37 @@
+  exit 1
+fi
+
+sleep 5 
```
What is this waiting for?

dr3 commented on can-i-merge/bash.sh:
```diff
@@ -0,0 +1,37 @@
+
+lastBuildOutput2=$(curl $LAST_BUILD_JOB_URL --cert $COSMOS_CERT --key $COSMOS_KEY)
+
+if [[ $lastBuildOutput2 =~ "nextBuild\":null" ]]; then
```
Would be better if this was functionalised to be less duplication

pjlee11 replied:
Tried that and it broke ... I'll try again 🌞

Commit:
change to a function and add explainer comment

Commit:
Remove double run as it's obsolete

Revisiting this and I've realised that the race condition will never be a factor. This is because I was comparing the trigger of a job against the checking of the API, whereas the instant check of the API is irrelevant given the fact github enforces we are up-to-date with latest.

Example:
PR 1 is merged at the same time as PR 2 is pushed. PR 2 triggers `can-i-merge` and is in the small timeframe that it will get a false-positive tick. Several seconds later Github enforces that the PR needs to pull in latest before merge and in doing so the `can-i-merge` job would be retriggered. Problem resolved.

Commit:
purposefully break test for testing reason

Commit:
debugging

Commit:
variables should be prefixed with $

jamesbhobbs commented:
PR closed and reopened to trigger Travis PR build

jamesbhobbs commented:
Second thoughts I'm just closing this one.
